tools/README.md

```diff
@@ -0,0 +1,34 @@
+# Tools
+
+The scripts in this folder help with commands with long arguments
+
+* generate_llvm_coverage.sh **(RECOMMENDED)**
+
+Generate coverage using the llvm coverage tool. The output files are generated to reports/llvm/html/index.html, reports/llvm/lcov.info & reports/llvm/lcov/index.html
+
+``` bash
+$ sh tools/generate_llvm_coverage.sh 
+... 
+Writing directory view page.
+Overall coverage rate:
+  lines......: 78.1% (890 of 1140 lines)
+  functions..: 55.1% (201 of 365 functions)
```

Comment:
Wowzers, there's a lot more functions captured by llvm coverage. Can you attach a sample report of each to this PR and maybe speak to some of the differences in functions inspected in the PR summary?

Reply:
we did notice a disparity between the tools, and some of the coverage has to do with _how_ the coverage is highlighted. Also the TRACE/DEBUG/WARN macros were not ignored in tarpaulin as they are with llvm - which could be the line coverage. 

Like any tool, the coverage needs an educated dev to assess the results given.

Reply:
Thanks for updating with the report. I see entries that are captured for `utils/`:

![image](https://github.com/user-attachments/assets/4b03d292-f589-4d74-9e1f-ed5a1ef114c4)

Could you check for a way to exclude that folder? It contains only utils code for making the examples and integration test infra.

Reply:
yes good call, i did it in tarpaulin. it should be available in llvm

Reply:
![image](https://github.com/user-attachments/assets/91b8dca9-86e6-4715-8b06-f17a7d5fbe03)
 no more utils

Reply:
All right, looks good!